src/app/components/AllowanceMenu/index.tsx

```diff
@@ -65,13 +104,31 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
     setIsOpen(false);
   }
 
+  function changedPermissionIds(): number[] {
+    if (!permissions || !originalPermissions) return [];
+    const ids = permissions
+      .filter((prm, i) => prm.enabled !== originalPermissions[i].enabled)
+      .map((prm) => prm.id);
+    return ids;
+  }
+
   async function updateAllowance() {
     await msg.request("updateAllowance", {
       id: allowance.id,
       totalBudget: parseInt(budget),
       lnurlAuth,
     });
 
+    const changedIds = changedPermissionIds();
+
+    if (changedIds.length) {
+      await msg.request("togglePermissions", {
+        ids: changedIds,
+      });
+    }
+
+    setOriginalPermissions(permissions);
```

Comment:
I would add a small comment here:
`// DB is updated, let´s update the original permissions to the updated permission in local state too`